pandas/tests/extension/base/getitem.py

```diff
@@ -158,21 +158,21 @@ def test_getitem_boolean_array_mask(self, data):
         result = pd.Series(data)[mask]
         self.assert_series_equal(result, expected)
 
-    def test_getitem_boolean_array_mask_raises(self, data):
+    def test_getitem_boolean_na_treated_as_false(self, data):
         mask = pd.array(np.zeros(data.shape, dtype="bool"), dtype="boolean")
```

Comment:
Can you take here something with True's as well? (now it will give an empty result)

Reply:
it would be good to run the test also for both a boolean array and a list as mask (to ensure the list works)

Reply:
Looks like the list input may not be working properly, will work on fixing that

Reply:
@jorisvandenbossche So I think the issue with list masks containing bools and pd.NA was that `is_bool_indexer` from `pandas/core/common.py` didn't consider these to be valid boolean indexers because it was trying to cast to a bool numpy array: https://github.com/pandas-dev/pandas/blob/master/pandas/core/common.py#L142

Made an update there to recognize pd.NA and also updated the test; hopefully CI will still pass. The assumption that boolean indexers are ones that can be cast as _numpy_ boolean arrays seems to happen in a lot of places (e.g., https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/base.py#L4147) so I could see this causing problems.

Reply:
Hmm, the "old" code of still accepting object dtype makes this a bit more complex indeed. Maybe instead of casting to a numpy array, we could use `pd.array`, so it will handle the case with boolean values better?

Reply:
I think that should theoretically work in combination with the right change to `is_bool_indexer`